### ISSUE TITLE
feature request: support local and depth arguments for :hook

### ISSUE THREAD

**Issue by Hugo-Heagren** [jwiegley/use-package#965]
I'm trying to port some of my old config into neater use-package declarations and I have a few `add-hook`s  which use the optional  `DEPTH` and `LOCAL` args of that function. These aren't currently supported by use-package, but it would be nice if they were. Perhaps something like one of these?

More consistent with use-package's use of keys:
```elisp
(use-package foo
  :hook
  (bar-mode . foo-mode-hook :depth 80 :local t))
```

And more consistent with `add-hook` (though with the disadvantage that comes with all optional positional arguments -- specifying only the last one requires specifying the others as nil, which is kinda cluttered):

```elisp
(use-package foo
  :hook
  (bar-mode . foo-mode-hook 80 t))
```

**Comment by conao3** [comment]
First of all, your grammar is invalid as Elisp
```
'(use-package foo
  :hook
  (bar-mode . foo-mode-hook :depth 80 :local t))
;;=> Debugger entered--Lisp error: (invalid-read-syntax ". in wrong context")
;;     read(#<buffer *scratch*>)
;;     elisp--preceding-sexp()
;;     elisp--eval-last-sexp(t)
;;     eval-last-sexp(t)
;;     eval-print-last-sexp(nil)
;;     funcall-interactively(eval-print-last-sexp nil)
;;     call-interactively(eval-print-last-sexp nil nil)
;;     command-execute(eval-print-last-sexp)
```

**Comment by Hugo-Heagren** [comment]
@conao3 you're right. I was rembering that it's possible to have many elements to the left of the `.`, and getting confused. Perhaps it could be done without cons cells, and just with a list, like this:

```elisp
(use-package foo
  :hook
  (bar-mode foo-mode-hook :depth 80 :local t))
```

or this:

```elisp
(use-package foo
  :hook
  (bar-mode foo-mode-hook 80 t))
```

Of course, for backwards compatibility the cons cell format could still be supported, it just wouldn't support the extra arguments. Or another way perhaps?

**Comment by jdtsmith** [comment]
This would be very helpful to have.  There is no clean way to combine defer and hook addition at a non-default depth.

**Comment by Hugo-Heagren** [comment]
@jdtsmith since making this request I have migrated to [leaf](https://github.com/conao3/leaf.el). One of my reasons was that the maintainer seems much more active on the project, and requests like this (and my PR on use-package) are more likely to be acted upon. I just haven't got round to making them yet...

**Comment by m-vitale** [comment]
Any hope for this? I think there's an increasing push for additional features that people have found lacking (see discussion in https://lists.gnu.org/archive/html/emacs-devel/2025-07/msg00354.html as an example).
I myself have a wrapper around use-package for various reasons and I'll be adding support for this as I don't like having things like:
```
  :config
  ;; TODO: neo/use-package could support dept in :hook
  (add-hook 'after-init-hook 'neo/desktop-restore 99) ; :hook doesn't support depth
```
but if it were supported by use-package it could benefit more people.
